ArduCopter/events.cpp

```diff
@@ -286,6 +286,36 @@ void Copter::failsafe_terrain_on_event()
     }
 }
 
+// check for tether fault
+void Copter::failsafe_tether_status_check()
+{
+    bool tether_trigger = copter.planck_interface.get_spool_status() == PLANCK_DECK_SPOOL_RETRIEVE;
+
+    if (tether_trigger != failsafe.tether) {
+        if (tether_trigger) {
+            failsafe_tether_on_event();
+        } else {
+            AP::logger().Write_Error(LogErrorSubsystem::FAILSAFE_TETHER, LogErrorCode::ERROR_RESOLVED);
+            failsafe.terrain = false;
+        }
+    }
+}
+
+// tether failsafe action
+// TODO: Make this handle/react differently to diffrent tether faults
+void Copter::failsafe_tether_on_event()
+{
+    failsafe.tether = true;
```

Comment:
What clears this?  Also, shouldn't this be set in the check method?

Reply:
It will clear when failsafe.tether = true and the tether status is not RETRIEVE_ONLY. I used the same structure as tge terrain failsafe, which has the "failsafe.terrain = true" inside the failsafe_terrain_on_event() method.

Reply:
OK. Kinda weird that setting the `failsafe.tether` happens in two separate methods, but its fine, I suppose.